src/components/LessonCard/LessonCard.tsx

```diff
@@ -108,63 +76,192 @@ const Paragraph = styled(P)`
+        </svg>
+      );
+  }
+}
+
+type MascotType = 1|2|3|4|5;
+type Progress = 0|1|2|3;
+
+type UnderlyingCardProps = {
+  isResourcesCard: boolean;
   lessonId: string;
+  mascotType: MascotType;
+  progress: Progress;
+  subtitle?: string;
+  title: string;
+  description: string;
+};
+
+type LessonCardProps = {
+  lessonId: string;
+  mascotType: MascotType;
+  progress: Progress;
   subtitle?: string;
   title: string;
   description: string;
-  call2actionText?: string;
+};
+
+type ResourcesCardProps = {
+  lessonId: string;
   mascotType: MascotType;
+  subtitle?: string;
+  title: string;
+  description: string;
 };
 
-export const LessonCard: FC<Props> = ({
+const UnderlyingCard: FC<UnderlyingCardProps> = ({
+  isResourcesCard,
   lessonId,
+  mascotType,
+  progress,
   subtitle,
   title,
   description,
-  call2actionText,
-  mascotType,
 }) => {
-  let mascotUrl;
-  switch (mascotType) {
-    case "lesson-1":
-      mascotUrl = shapesMascotLesson1Url;
-      break;
-    case "lesson-2":
-      mascotUrl = shapesMascotLesson2Url;
-      break;
-    case "lesson-3":
-      mascotUrl = shapesMascotLesson3Url;
-      break;
-    case "lesson-4":
-      mascotUrl = shapesMascotLesson4Url;
-      break;
-    case "lesson-5":
-      mascotUrl = shapesMascotLesson5Url;
-      break;
-    default:
-      mascotUrl = shapesMascotLesson1Url;
-      break;
-  }
   return (
     <Wrap>
+
       <BlockMascot $mascotType={mascotType}>
-        <ImageMascot src={mascotUrl} />
+        <DivMascot>
+          {selectMascot(mascotType, isResourcesCard, progress)}
+        </DivMascot>
+        {!isResourcesCard && (
+          <DivProgress>
+            <img src={urlGearSvg} />
+            <span>{progress}/3</span>
+          </DivProgress>
+        )}
```

Comment:
I would advice not to have a boolean or enum input prop to distinguish different implementation of a sharable component. It's against the principle of sharable component where it should have not kept knowledge of consumer specific implementations.

In this specific case, progress widget is only needed for resources card. Instead of having a `isResourcesCard` prop, we can replace it with a `progress` prop

```tsx
type UnderlyingCardProps = {
    ...
    progress: [number, number]
    ...
}

...
{!progress && (
    <DivProgress>
    <img src={urlGearSvg} />
    <span>{progress[0]}/{progress[1]}</span>
    </DivProgress>
)}

```

Then it's no longer about resouces card, but progress data, which is generic enough to be sharable for any component that needs a progress widget.

There is also a more flexible alternative approach:

We can have an empty slot injected in underlaying card and leave it to the consumer to decide what to render there.

```tsx
...
</DivMascot>
{children}
</BlockMascot>
...
```

Then the implementation of progress widget can be handed over as ResourcesCard's responsibility

```tsx
export const ResourcesCard = () => {
    return <UnderlayingCard {...}>
      <DivProgress>
        <img src={...} />
        <span>?/?</span>
        </DivProgress>
    </UnderlyingCard>
};
```

Reply:
I appreciate that @owen26 comment here is a feedback based on continuing down this path of "UnderlyingCard", but see my comment about the `isResourcesCard`. I just don't think it is appropriate or necessary to introduce a distinct variant of this LessonCard, when as I originally envisioned it, it's simply a LessonCard, it's just it has optional bits that can be turned on/off.